src/Workspaces/Core/Portable/EmbeddedLanguages/VirtualChars/VirtualCharSequence.Chunks.cs

```diff
@@ -67,9 +68,16 @@ public StringChunk(int firstVirtualCharPosition, string data)
             public override int Length => _underlyingData.Length;
```

Comment:
❔ Have we verified that the lengths exposed in the API accounts for cases (if necessary) where the number of code points in the string does not match the number of code units in the string?

Reply:
yes.  wil explain below.